django/db/migrations/state.py

```diff
@@ -443,6 +443,11 @@ def from_model(cls, model, exclude_rels=False):
                 elif name == "index_together":
                     it = model._meta.original_attrs["index_together"]
                     options[name] = set(normalize_together(it))
+                elif name == "indexes":
+                    indexes = model._meta.indexes
```

Comment:
I think the copying in `ModelBase.__new__()` is sufficient -- at least, I'm not sure what a failing test case would be to show why your suggested change is required.